app/scripts/models/auth_brokers/pairing/supplicant.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import OAuthRedirectBroker from '../oauth-redirect';
+import PairingChannelClient from '../../../lib/pairing-channel-client';
+import setRemoteMetaData from './remote-metadata';
+import SupplicantStateMachine from '../../pairing/supplicant-state-machine';
+import Url from '../../../lib/url';
+
+export default class SupplicantBroker extends OAuthRedirectBroker {
+  type = 'supplicant';
+
+  initialize (options = {}) {
+    super.initialize(options);
+
+    const { config, notifier, relier } = options;
+    const channelServerUri = config.pairingChannelServerUri;
+    const { channelId, channelKey } = relier.toJSON();
+    if (channelId && channelKey && channelServerUri) {
+      this.pairingChannelClient = new PairingChannelClient({
+        channelId,
+        channelKey,
+        channelServerUri,
+      },
+      {
+        importPairingChannel: options.importPairingChannel,
+        notifier,
+      });
+
+      this.suppStateMachine = new SupplicantStateMachine({}, {
+        broker: this,
+        notifier,
+        pairingChannelClient: this.pairingChannelClient,
+        relier
+      });
+
+      this.pairingChannelClient.open();
```

Comment:
This method returns a promise, do you need to propagate it? Does the broker care if there is a problem connecting to the pairing channel?

Reply:
this being inside the `initialize`, i didn't find the use for this yet